lib/whitehall.rb

```diff
@@ -212,6 +212,6 @@ def self.secrets_path
   private_class_method :secrets_path
 
   def self.integration_or_staging?
-    ENV.fetch("GOVUK_WEBSITE_ROOT") =~ /integration|staging/
+    ENV.fetch("GOVUK_WEBSITE_ROOT", "") =~ /integration|staging/
```

Comment:
I'm guessing this defaults to "" if `GOVUK_WEBSITE_ROOT` isn't defined.

That's probably a good idea 👍   just wondering if you added this because it fell over when deployed? If so, do we know that it actually works on integration?

Reply:
I _think_ it was actually that it falls over when testing it locally. There were no issues specifically with this when I deployed, it was just that I couldn't test the actual change without defining this.

Reply:
```
irb(main):001:0> Whitehall.integration_or_staging?
=> 12
```

An integer was not what I was expecting here, but an integer is truthy, I suppose.

Just to triple-check:

```
irb(main):003:0> ENV.fetch("GOVUK_WEBSITE_ROOT", "")
=> "https://www.integration.publishing.service.gov.uk"
…
irb(main):008:0>  ENV.fetch("NOT_A_REAL_VARIABLE", "") =~ /integration|staging/
=> nil
```

Reply:
Ha, I was surprised by that too! It really ought to return a boolean, as that would follow the `method_name_ending_with_question_mark?` convention. We could do so with:

```
    !(ENV.fetch("GOVUK_WEBSITE_ROOT", "") =~ /integration|staging/).nil?
```

Reply:
Yeah, presumably it's always going to be interpreted as a boolean anyway but it would be nicer to do so explicitly.

Reply:
`.present?` is nicer 👍